src/test/java/com/bobocode/hoverla/bring/context/BeanConfigClassValidatorTest.java

```diff
@@ -0,0 +1,101 @@
+import com.bobocode.hoverla.bring.testsubject.validator.config.EnumTestBeanConfig;
+import com.bobocode.hoverla.bring.testsubject.validator.config.InnerClassTestBeanConfig;
+import com.bobocode.hoverla.bring.testsubject.validator.config.InterfaceTestBeanConfig;
+import com.bobocode.hoverla.bring.testsubject.validator.config.InvalidBeanMethodsTestConfig;
+import com.bobocode.hoverla.bring.testsubject.validator.config.NoDefaultConstructorTestConfig;
+import com.bobocode.hoverla.bring.testsubject.validator.config.RecordTestBeanConfig;
+import com.bobocode.hoverla.bring.testsubject.validator.config.ValidTestBeanConfig;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+class BeanConfigClassValidatorTest {
+
+    private BeanConfigClassValidator validator;
+
+    @SneakyThrows
+    private static Stream<Arguments> invalidBeanConfigs() {
+        return Stream.of(
+                Arguments.of(
+                        AbstractTestBeanConfig.class,
+                        "Configuration class must not be an abstract class",
+                        "Doesn't allow bean config class to be abstract"),
+                Arguments.of(EnumTestBeanConfig.class,
+                        "Configuration class must not be an enum",
+                        "Doesn't allow bean config class to be an enum"),
+                Arguments.of(InterfaceTestBeanConfig.class,
+                        "Configuration class must not be an interface",
+                        "Doesn't allow bean config class to be an interface"),
+                Arguments.of(NoDefaultConstructorTestConfig.class,
+                        "Configuration class must have one public no-arguments constructor",
+                        "Doesn't allow bean config class with no public no-arg constructor"),
+                Arguments.of(Class.forName("com.bobocode.hoverla.bring.testsubject.validator.config.NonPublicTestConfig"),
+                        "Configuration class must be public",
+                        "Doesn't allow non-public bean config class"),
+                Arguments.of(RecordTestBeanConfig.class,
+                        "Configuration class must not be a record",
+                        "Doesn't allow bean config class to be a record"),
+                Arguments.of(InnerClassTestBeanConfig.InnerTestBeanConfig.class,
+                        "Configuration class must not have an enclosing class",
+                        "Doesn't allow bean config class to be inner or nested")
+        );
+    }
+
+    @SneakyThrows
```

Comment:
Why do you need `@SneakyThrows` here?

Reply:
Yeah, we can also use 
` throws ClassNotFoundException 
`
This line throws it, we can't just use class literal to get this non public class
`                Arguments.of(Class.forName("com.bobocode.hoverla.bring.testsubject.validator.config.NonPublicTestConfig")
`

Reply:
Nope) You actually need `@SneakyThrows` in `invalidBeanConfigs()` on line `29` because of `Class.forName()`. But why do you need it in `invalidBeanMethodsMessages()` on line `57`?

Reply:
Yeah, I used to need it, but then this code was removed, but not SneakyThrows. Good catch

Reply:
Fixed